include/mxnet/kvstore.h

```diff
@@ -387,6 +395,12 @@ class KVStore {
    */
   std::string type_;
 
+  /** \brief Gradient compression object starts with GC_NONE mode
+   * Used if SetGradientCompression sets the type.
+   * Currently there is no support for un-setting gradient compression
+   */
+  std::shared_ptr<kvstore::GradientCompression> gradient_compression_;
```

Comment:
`no support for un-setting gradient compression `? What happens if an user tries to unset it?

Reply:
if user uses `kvstore.set_gradient_compression({'type':'none'}` after setting it to `2bit`, it throws an error saying none can't be a type. 
If users sets `2bit` again with different threshold, then new threshold will be used, but there might be a period in transition when gradients quantized with old threshold will be dequantized with new threshold. This is because some operations may be queued up in the engine. So, calling `set_gradient_compression` twice isn't supported.